astropy/utils/data.py

```diff
@@ -1523,218 +1540,46 @@ def _get_download_cache_locs(pkgname='astropy'):
-                # gets in the way of useful error messages.
-                msg = (
-                    f"Cache is locked after {waited:.2f} s. This may indicate "
-                    f"an astropy bug or that kill -9 was used. If you want to "
-                    f"unlock the cache remove the directory {lockdir}.")
-            except OSError as e:
-                # PermissionError doesn't cover all read-only-ness, just EACCES
-                if e.errno in [errno.EPERM,    # Operation not permitted
-                               errno.EACCES,   # Permission denied
-                               errno.EROFS]:   # File system is read-only
-                    if need_write:
-                        raise
-                    else:
-                        break
-                else:
-                    raise
-            else:
-                got_lock = True
-                # write the pid of this process for informational purposes
-                with open(pidfn, 'w') as f:
-                    f.write(str(os.getpid()))
-                break
-        else:
-            # Never did get the lock.
+        datadir = os.path.join(astropy.config.paths.get_cache_dir(pkgname), 'download', 'url')
+
+        if not os.path.exists(datadir):
             try:
-                # Might as well try to read and report the PID file, at
-                # this point it's unlikely we'll block someone else trying
-                # to exit cleanly.
-                pid = get_file_contents(pidfn)
+                os.makedirs(datadir)
             except OSError:
-                pass
-            else:
-                msg += f" Lock claims to be held by process {pid}."
-            raise RuntimeError(msg)
+                if not os.path.exists(datadir):
+                    raise
+        elif not os.path.isdir(datadir):
+            raise OSError(f'Data cache directory {datadir} is not a directory')
 
-        yield
+        return datadir
+    except OSError as e:
+        msg = 'Remote data cache could not be accessed due to '
+        estr = '' if len(e.args) < 1 else (': ' + str(e))
+        warn(CacheMissingWarning(msg + e.__class__.__name__ + estr))
+        raise
 
-    finally:
-        # clear_download_cache might have deleted the lockdir
-        if got_lock and os.path.exists(lockdir):
-            if os.path.isdir(lockdir):
-                # if the pid file is present, be sure to remove it
-                if os.path.exists(pidfn):
-                    os.remove(pidfn)
-                os.rmdir(lockdir)
-            else:
-                raise RuntimeError(
-                    f'Error releasing lock. {lockdir} exists but is not '
-                    f'a directory.')
-        else:
-            # Just in case we were called from _clear_download_cache
-            # or something went wrong before creating the directory
-            # or the cache was readonly; no need to clean it up then.
-            pass
+
+def _url_to_dirname(url):
```

Comment:
Isn't this more accurate?
```suggestion
def _url_to_hash(url):
```

Reply:
No, or not exactly - this function is intended to extract the name of the storage directory for a URL. At the moment that operation is hashing, but I could see adding, say, a prefix so that dirnames were more easily globbed.